src/main/java/uk/nhs/tis/trainee/notifications/event/ProgrammeMembershipListener.java

```diff
@@ -54,7 +59,16 @@ public ProgrammeMembershipListener(ProgrammeMembershipService programmeMembershi
   @SqsListener("${application.queues.programme-membership}")
   public void handleProgrammeMembershipUpdate(ProgrammeMembershipEvent event) {
     log.info("Handling programme membership update event {}.", event);
-    boolean isExcluded = programmeMembershipService.isExcluded(event);
-    log.info("Programme membership {}: excluded {}.", event.tisId(), isExcluded);
+    if (event.recrd() != null && event.recrd().getData() != null) {
+      try {
+        ProgrammeMembership programmeMembership = mapper.toEntity(event.recrd().getData());
+        boolean isExcluded = programmeMembershipService.isExcluded(programmeMembership);
+        log.info("Programme membership {}: excluded {}.", event.tisId(), isExcluded);
+      } catch (JsonProcessingException e) {
+        log.error("Invalid programme membership event: {}", event);
```

Comment:
I guess this is debatable: would we rather allow this to requeued so it will end up on the DLQ?